File: 02/js/PriorityQueue.js
const MaxHeap = require('./MaxHeap');

module.exports = class {
    constructor() {
        this.maxHeap = new MaxHeap();
    }

    getFront () {
        return this.maxHeap.findMax();
    }

    getSize() {
        return this.maxHeap.size();
    }

    isEmpty() {
        return this.maxHeap.isEmpty();
    }

    enqueue(elm) {
        this.maxHeap.add(elm);
    }

    dequeue() {
        return this.maxHeap.extractMax();
    }
};

File: 02/java/PriorityQueue.java
public class PriorityQueue<E extends Comparable<E>> implements Queue<E> {
    private MaxHeap<E> maxHeap;

    public PriorityQueue() {
        maxHeap = new MaxHeap<>();
    }

    public E getFront() {
        return maxHeap.findMax();
    };

    public int getSize() {
        return maxHeap.size();
    };

    public boolean isEmpty() {
        return maxHeap.isEmpty();
    };

    public void enqueue(E e) {
        maxHeap.add(e);
    };

    public E dequeue() {
        return maxHeap.extractMax();
    };
}